Sources/MapboxCoreNavigation/NavigationEventsManager.swift

```diff
@@ -387,17 +413,30 @@ open class NavigationEventsManager {
             sendOutstandingFeedbackEvents(forceAll: false)
         }
         
-        if sessionState?.arrivalTimestamp == nil,
+        if sessionState.arrivalTimestamp == nil,
             progress.currentLegProgress.userHasArrivedAtWaypoint {
-            sessionState?.arrivalTimestamp = dataSource?.router.location?.timestamp ?? Date()
+            sessionState.arrivalTimestamp = activeNavigationDataSource?.router.location?.timestamp ?? Date()
             sendArriveEvent()
             
             return
         }
         
-        if sessionState?.departureTimestamp == nil {
-            sessionState?.departureTimestamp = dataSource?.router.location?.timestamp ?? Date()
+        if sessionState.departureTimestamp == nil {
+            sessionState.departureTimestamp = activeNavigationDataSource?.router.location?.timestamp ?? Date()
             sendDepartEvent()
         }
     }
+    
+    func incrementDistanceTraveled(by distance: CLLocationDistance) {
+        sessionState.totalDistanceCompleted += distance
+    }
+    
+    func arriveAtWaypoint() {
+        sessionState.departureTimestamp = nil
+        sessionState.arrivalTimestamp = nil
+    }
+    
+    func record(locations: [CLLocation]) {
```

Comment:
Hide the argument’s label using `_`, since at the point of use, `record(locations: locations)` would be redundant.